Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -1,62 +1,89 @@
 package com.gildedrose;
 
 class GildedRose {
+
+    public static final String SULFURAS = "Sulfuras, Hand of Ragnaros";
+    public static final String CHEESE = "Aged Brie";
+    //TODO: brainstorm the naming a bit here.
+    public static final String BACKSTAGE_PASSES_TO_A_TAFKAL_80_ETC_CONCERT
+        = "Backstage passes to a TAFKAL80ETC concert";
+
     Item[] items;
 
     public GildedRose(Item[] items) {
         this.items = items;
     }
 
+    /**
+     * Update the staet of the items.
+     */
     public void updateQuality() {
-        for (int i = 0; i < items.length; i++) {
-            if (!items[i].name.equals("Aged Brie")
-                    && !items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                if (items[i].quality > 0) {
-                    if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                        items[i].quality = items[i].quality - 1;
+        for (Item item : items) {
+            if (!item.name.equals(CHEESE)
```

Comment:
The functionality branches a lot, which makes it hard to read. Ideally each item types functionality would be encapsulated.